MPOL_LOCAL "not declared" when compiling C program

Hi!

I've been trying to use `MPOL_LOCAL` as the `mode` to `mbind` but the symbol is not defined.

```c
#include <numaif.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>    

#define N 134217728

int main() {
    uint64_t *a = (uint64_t*) malloc(N*sizeof(uint64_t));
    mbind(a, N, MPOL_LOCAL, 0, 0, MPOL_MF_STRICT | MPOL_MF_MOVE);
    printf("Hello world!\n");
    return 0;
}
```

```c
$ gcc-8 -lnuma example.c
example.c: In function ‘main’:
example.c:10:14: error: ‘MPOL_LOCAL’ undeclared (first use in this function); did you mean ‘MPOL_MAX’?
  mbind(a, N, MPOL_LOCAL, 0, 0, MPOL_MF_STRICT | MPOL_MF_MOVE);
              ^~~~~~~~~~
              MPOL_MAX
example.c:10:14: note: each undeclared identifier is reported only once for each function it appears in
```

https://stackoverflow.com/q/52632754/4288486

--- a/numaif.h
+++ b/numaif.h
@@ -26,9 +26,12 @@ extern long move_pages(int pid, unsigned long count,
 #define MPOL_PREFERRED   1
 #define MPOL_BIND        2
 #define MPOL_INTERLEAVE  3
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
 #define MPOL_LOCAL       4
-
-#define MPOL_MAX MPOL_INTERLEAVE
+#define MPOL_MAX         5
+#else
+#define MPOL_MAX         4
+#endif
 
 /* Flags for get_mem_policy */
 #define MPOL_F_NODE    (1<<0)   /* return next il node or node of address */

--- a/numaif.h
+++ b/numaif.h
@@ -1,8 +1,6 @@
 #ifndef NUMAIF_H
 #define NUMAIF_H 1
 
-#include <linux/version.h>
-
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -28,12 +26,8 @@ extern long move_pages(int pid, unsigned long count,
 #define MPOL_PREFERRED   1
 #define MPOL_BIND        2
 #define MPOL_INTERLEAVE  3
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
 #define MPOL_LOCAL       4
 #define MPOL_MAX         5
-#else
-#define MPOL_MAX         4
-#endif
 
 /* Flags for get_mem_policy */
 #define MPOL_F_NODE    (1<<0)   /* return next il node or node of address */

--- a/numaif.h
+++ b/numaif.h
@@ -1,6 +1,8 @@
 #ifndef NUMAIF_H
 #define NUMAIF_H 1
 
+#include <linux/version.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif

--- a/numaif.h
+++ b/numaif.h
@@ -23,9 +23,10 @@ extern long move_pages(int pid, unsigned long count,
 
 /* Policies */
 #define MPOL_DEFAULT     0
-#define MPOL_PREFERRED    1
+#define MPOL_PREFERRED   1
 #define MPOL_BIND        2
 #define MPOL_INTERLEAVE  3
+#define MPOL_LOCAL       4
 
 #define MPOL_MAX MPOL_INTERLEAVE
 

--- a/test/randmap.c
+++ b/test/randmap.c
@@ -105,7 +105,7 @@ void testmap(void)
 
 	for (;;) {
 		unsigned long offset = random() % PAGES;
-		int policy = random() % (MPOL_MAX+1);
+		int policy = random() % (MPOL_MAX);
 		unsigned long nodes = random() % 4;
 		long length = random() % (PAGES - offset);
 

